src/providers/core_provider/mod.rs

```diff
@@ -32,8 +40,11 @@ pub struct CoreProvider {
     wire_protocol_version_maj: u8,
     provider_info: Vec<ProviderInfo>,
     provider_opcodes: HashMap<ProviderID, HashSet<Opcode>>,
+    authenticator_info: Vec<AuthenticatorInfo>,
     #[derivative(Debug = "ignore")]
     prov_list: Vec<Arc<dyn Provide + Send + Sync>>,
+    #[derivative(Debug = "ignore")]
+    auth_list: Vec<Arc<dyn Authenticate + Send + Sync>>,
```

Comment:
I don't think you need this here, you only need the `authenticator_info` above - that makes the process simpler, as you don't have to make ownership of authenticators shared between the `CoreProvider` and the `FrontEndHandler`.

The `provider_info` example is a bit skewed now because we actually need an interface to the other providers for the upcoming operations. Previously it only took in a vector of `ProviderInfo`

Reply:
Do you think there might be a case for having this in the future? I agree that it would be much simpler without it, but at the same time, I was thinking that there might be a situation in the future where the core provider might need access to the authenticators.

If you can't see a future where this is needed then I'll gladly simplify! 👍

Reply:
It probably will at some point (e.g. for dynamically setting the public key of an Identity Provider or stuff like that), but I don't know if it's good practice to keep it in here for now. I'll defer this to @hug-dev to express his thoughts on this too

Reply:
I would say it is better to remove it for now if it is not needed. I agree that it might be used in the future but I think adding it in the future might not be that hard.

Looking at the different fields of the `CoreProvider` structures, could we simplify though? For example can the `provider_info` and `provider_opcodes` data be directly computed from `prov_list`? Maybe it is similar for `authenticator_info` and `auth_list`. This is not to change in this PR anyway but something to remember.

Reply:
Okay -- I've changed this now.

Note that this will still fail CI because of versioning woes. I will wait until @sbailey-arm's operations are merged (which includes a version bump) then I will rebase on top of that. 🙂

Reply:
What new versioning do you need? I believe that the client was published with new versions of everything including ListAuthenticators

Reply:
Oh I see what you mean, ignore my comment 😅